### ISSUE TITLE
jwk construct with hs256 returns 'k' as binary

### ISSUE THREAD

**Issue by piedrahitapablo** [mpdavis/python-jose#137]
Not sure if this is expected behavior or a minor bug, but when `jwk.construct` is used and the jwk is converted to a dict using `.to_dict`, the 'k' attribute is returned as a binary string. This is not an issue until the dict is used to decode a JWT and an AttributeError is raised because 'k' does not have a `encode` method ('bytes' object has no attribute 'encode'). To reproduce:

```
from jose import jwk, jwt

user_jwt = jwt.encode({'username': 'user'}, 'secret')

jwk_obj = jwk.construct('secret', 'HS256')
jwk_dict = jwk_obj.to_dict()

# to avoid the exception:
# jwk_dict['k'] = jwk_dict['k'].decode('utf-8')

claims = jwt.decode(user_jwt, jwk_dict)
```

**Comment by zejn** [comment]
This is indeed a bug,

**Comment by piedrahitapablo** [comment]
Is there any way that I can help to solve this? I don't know much about this topic but two possible solutions would be i) change the `to_dict` method to decode the key into a regular string or ii) change the necessary code in `decode` to accept binaries as well.

**Comment by blag** [comment]
The first option:

> i) change the `to_dict` method to decode the key into a regular string

makes the most sense to me.

**Comment by blag** [comment]
Also reported in #127. Fixed in #139.
